src/modules/mod_magma2_cfortran.F

```diff
@@ -0,0 +1,230 @@
+implicit none
+
+!! =============================================================================
+!! Fortran interfaces to C functions
+interface
+
+    !! -------------------------------------------------------------------------
+    !! CPU interfaces (matrix in CPU memory)
+    subroutine magma_cgetrf( m, n, A, lda, ipiv, info ) &
+    bind(C, name="magma_cgetrf")
+        use iso_c_binding
+        integer(c_int),            value  :: m, n, lda
+        complex(c_float_complex), target :: A(lda,*)
+        integer(c_int),            target :: ipiv(*)
+        integer(c_int),            target :: info  !! int*
+    end subroutine
+
+    subroutine magma_cpotrf( uplo, n, A, lda, info ) &
+    bind(C, name="magma_cpotrf")
+        use iso_c_binding
+        integer(c_int),            value  :: uplo
+        integer(c_int),            value  :: n, lda
+        complex(c_float_complex), target :: A(lda,*)
+        integer(c_int),            target :: info  !! int*
+    end subroutine
+
+    subroutine magma_cgeev( jobvl, jobvr, n, A, lda, w, VL, ldvl, VR, ldvr, &
+                            work,  lwork, rwork, info) &
+    bind(C, name="magma_cgeev")
+        use iso_c_binding
+        integer(c_int),            value  :: jobvl, jobvr
+        integer(c_int),            value  :: n, lda, ldvl, ldvr, lwork
+        complex(c_float_complex), target :: A(lda,*)
+        complex(c_float_complex), target :: w(*)
+        complex(c_float_complex), target :: VR(ldvr,*),  VL(ldvl,*)
+        complex(c_float_complex), target :: work(*)
+        real(c_float),            target :: rwork(*)
+        integer(c_int),            target :: info  !! int*
+    end subroutine
+
+    subroutine magma_cgeev_m( jobvl, jobvr, n, A, lda, w, VL, ldvl, VR, ldvr, &
+                            work,  lwork, rwork, info) &
+    bind(C, name="magma_cgeev_m")
+        use iso_c_binding
+        integer(c_int),            value  :: jobvl, jobvr
+        integer(c_int),            value  :: n, lda, ldvl, ldvr, lwork
+        complex(c_float_complex), target :: A(lda,*)
+        complex(c_float_complex), target :: w(*)
+        complex(c_float_complex), target :: VR(ldvr,*),  VL(ldvl,*)
+        complex(c_float_complex), target :: work(*)
+        real(c_float),            target :: rwork(*)
+        integer(c_int),            target :: info  !! int*
+    end subroutine
+
+    subroutine magma_cheevd_m( ngpu, jobz, uplo, n, A, lda, w, work,  lwork, &
+                               rwork, lrwork, iwork, liwork, info) &
+    bind(C, name="magma_cheevd_m")
+        use iso_c_binding
+        integer(c_int),            value  :: ngpu
+        integer(c_int),            value  :: jobz, uplo
+        integer(c_int),            value  :: n, lda, lwork, lrwork, liwork
+        complex(c_float_complex), target :: A(lda,*)
+        real(c_double),            target :: w(*)
+        complex(c_float_complex), target :: work(*)
+        real(c_double),            target :: rwork(*)
```

Comment:
In the MAGMA Fortran stubs `w` and `rwork` are float, not double. Any reason they are double in here?

Reply:
Hi,
you are right, this looks like a mistake... [ I haven't written the modules myself, but generated from magma-2.7.1 using a codegen.py script (only the z version is there)... Here the point is that the script is no longer distributed with this version of magma and I've downloaded a previous version... ]

nevertheless, you are right and this needs to be fixed !
I've attempted a fix here: 20369f33b04a8a2c8fbc18bda64fe394ccbe08cf